### PR TITLE
Fix file URLs with digest

### LINKED ISSUE #1217: Using file:/// URLs with digest fails due at digest check
### Description

Reproducer (hit this while building ppc64le iso and then trying to use - but this reproducer doesn't need that_

1. wget -P /tmp https://cloud.debian.org/images/cloud/bullseye/20221108-1193/debian-11-generic-amd64-20221108-1193.qcow2
2. Replace location: with "file:///tmp/debian-11-generic-amd64-20221108-1193.qcow2" in examples/debian.yaml
3. ./_output/bin/limactl --debug start ./examples/debian.yaml

The fix is easy in `pkg/downloader/download.go` I made the `validateLocalFileDigest` in `copyLocal` use the canonical local path `srcPath` not `src` which is a file URL.

Happy to submit a PR, but wanted to check about testing as currentlyonly the remote path is currently tested in `download_test.go`

```
DEBU[0000] interpreting argument "./examples/debian.yaml" as a file path for instance "debian"
? Creating an instance "debian" Proceed with the current configuration
DEBU[0001] Make sure "shared" network is stopped
DEBU[0001] Make sure "bridged" network is stopped
DEBU[0001] Make sure "host" network is stopped
INFO[0001] Attempting to download the image from "file:///tmp/debian-11-generic-amd64-20221108-1193.qcow2"  digest="sha512:c24edc82a8e3c9764d39b4b45cd8ad3d48f2927fc3b99431dde4ff69de8098d5f824fec570a520d38f1f95dd1f32c630803a473253f7d10a20de7dd3c09ccca0"
DEBU[0001] verifying digest of local file "file:///tmp/debian-11-generic-amd64-20221108-1193.qcow2" (sha512:c24edc82a8e3c9764d39b4b45cd8ad3d48f2927fc3b99431dde4ff69de8098d5f824fec570a520d38f1f95dd1f32c630803a473253f7d10a20de7dd3c09ccca0)
FATA[0001] failed to download the image, attempted 1 candidates, errors=[failed to download "file:///tmp/debian-11-generic-amd64-20221108-1193.qcow2": open file:///tmp/debian-11-generic-amd64-20221108-1193.qcow2: no such file or directory]
```

### PR DESCRIPTION
**Pull request by pnasrat**
Fixes #1217

Signed-off-by: Pris Nasrat <pnasrat@gmail.com>

### CODE REVIEW

**Comment by pnasrat** [comment]
Ok that Windows test looks relevant, but possibly a issue with go on windows eg golang/go#51442

I'll see if I can figure out.

```
  testing.go:1097: TempDir RemoveAll cleanup: remove C:\users\runner\Temp\TestDownloadLocalwith_digest942769228\002\empty.iso: Sharing violation.
```

File: pkg/downloader/downloader_test.go [hunk 1]
```diff
@@ -80,3 +81,39 @@ func TestDownloadRemote(t *testing.T) {
...
 	})
 }
+
+func TestDownloadLocal(t *testing.T) {
+
+	if testing.Short() {
+		t.Skip()
+	}
+
+	const emptyFileDigest = "sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"
```
Comment (AkihiroSuda) [inline pkg/downloader/downloader_test.go:91]: Can we use non-empty value, eg..,
```console
$ echo -n "TestDownloadLocal" | sha256sum
0c1e0fba69e8919b306d030bf491e3e0c46cf0a8140ff5d7516ba3a83cbea5b3  -
```

File: pkg/downloader/downloader_test.go [hunk 2]
```diff
@@ -80,3 +81,39 @@ func TestDownloadRemote(t *testing.T) {
 		assert.Equal(t, StatusUsedCache, r.Status)
 	})
 }
+
+func TestDownloadLocal(t *testing.T) {
+
+	if testing.Short() {
+		t.Skip()
```
Comment (AkihiroSuda) [inline pkg/downloader/downloader_test.go:88]: No need to skip. The test should be short enough.

File: pkg/downloader/downloader_test.go [hunk 3]
```diff
@@ -80,3 +81,39 @@ func TestDownloadRemote(t *testing.T) {
...
+	}
+
+	const emptyFileDigest = "sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"
+
+	t.Run("without digest", func(t *testing.T) {
+		localPath := filepath.Join(t.TempDir(), t.Name())
+		localFile := filepath.Join(t.TempDir(), "empty.iso")
```
Comment (AkihiroSuda) [inline pkg/downloader/downloader_test.go:95]: No need to add `.iso` suffix
```suggestion
		localFile := filepath.Join(t.TempDir(), "some-file")
```

**Comment by AkihiroSuda** [comment]
Thanks, windows test can be skipped

File: pkg/downloader/downloader_test.go [hunk 4]
```diff
@@ -80,3 +82,37 @@ func TestDownloadRemote(t *testing.T) {
 		assert.Equal(t, StatusUsedCache, r.Status)
 	})
 }
+
+func TestDownloadLocal(t *testing.T) {
```
Comment (AkihiroSuda) [inline pkg/downloader/downloader_test.go:86]:
```go
if runtime.GOOS == "windows" {
  // FIXME: `TempDir RemoveAll cleanup: remove C:\users\runner\Temp\TestDownloadLocalwithout_digest2738386858\002\test-file: Sharing violation.`
  t.Skip("Skipping on windows")
}
```

Comment (pnasrat) [reply pkg/downloader/downloader_test.go:87]: Done. I took a quick look, on how the windows test run via wine and will try figure how to fix/workaround. Sorry for the delay on this PR my end - the end of the year became very busy for me

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**